### ISSUE TITLE
Issue with SmartPension

### ISSUE THREAD

**Issue by Frenchcooc** [Bearer/Pizzly#75]
_This is a bug reported by @Hagbarth . I have not try to reproduce it yet. Just writing what he's shared with me_

## How to reproduce:

1. Add thee SmartPension API with the following config:

   ```
   {
    "name": "SmartPension",
    "auth": {
        "authType": "OAUTH2",
        "bodyFormat": "json",
        "grantType": "authorization_code",
        "config": {
        "scope": ["customer"]
        },
        "tokenParams": {},
        "tokenURL": "https://id.autoenrolment.co.uk/oauth/token",
        "authorizationMethod": "body",
        "authorizationParams": {
        "responseType": "code"
        },
        "authorizationURL": "https://id.autoenrolment.co.uk/oauth/authorize"
    },
    "request": {
        "headers": {
        "Accept": "application/json",
        "User-Agent": "Pizzly",
        "Authorization": "Bearer ${auth.accessToken}"
        },
        "baseURL": "https://api.autoenrolment.co.uk"
    }
   }
   ```

2. Create an OAuth application at SmartPension website ([link](https://partner.autoenrolment.co.uk/partners/sign-up))
3. Configure the API in your Pizzly dashboard
4. Try to connect

**Comment by Hagbarth** [comment]
For context this is what I'm seeing:
![ezgif-2-304885b853d8](https://user-images.githubusercontent.com/2462366/87145361-7f58a980-c2a9-11ea-941a-6b0124e04983.gif)


As you can see, I'm taking no action in the popup before the error message appears.

**Comment by Hagbarth** [comment]
Error from the console:

```
Error: Authorization failed. The response type is not supported
    js latest/dist/index.umd.min.js:1
connect:100:21
    <anonymous> https://pizzly.pento.fun/dashboard/smart-pension-dev/authentications/connect:100
```

**Comment by Frenchcooc** [comment]
Hi @Hagbarth! I believe this issue comes from the fact that SmartPension consent screen uses `postMessage` to send event to the parent window. They probably use that in their own authentication scenario, but that interfers with Pizzly!

I've made changes in the PR #76 and have released a new version of pizzly-js (`v0.2.6`). Can you update your version and try again? This should work as expected now.

**Comment by joaodias** [comment]
Hi @Frenchcooc! I'm working with @Hagbarth on this. The `response type is not supported error is gone` 🙌 

However, we now get a new error after the modal closes:

```
Authorization failed. The authorization modal sent an unsupported MessageEvent.
```

Do you have any thoughts on why this is happening?

**Comment by Frenchcooc** [comment]
Hi @joaodia! Thanks for reporting this issue. I've introduced a bug by fixing the previous issue... Latest PR #77 fix the bug and a new version of pizzly-js has been published (`v0.2.7`). I'm really sorry for the trouble caused here.

**Comment by joaodias** [comment]
@Frenchcooc no problem! Thanks for fixing so fast ⚡ 

I confirm it's fixed now 🙌 

Closing the issue
